go-runtime/encoding/encoding_test.go

```diff
@@ -134,17 +135,15 @@ func TestUnmarshal(t *testing.T) {
 		{name: "UnregisteredSumType", input: `{"d":{"name":"Variant","value":{"message":"hello"}}}`, expected: struct{ D unregistered }{}, err: `the only supported interface types are enums or any, not encoding_test.unregistered`},
 	}
 
-	tr := reflection.NewTypeRegistry()
-	tr.RegisterSumType(reflect.TypeFor[discriminator](), map[string]reflect.Type{
+	reflection.RegisterSumType(reflect.TypeFor[discriminator](), map[string]reflect.Type{
```

Comment:
We should probably add a `reflection.ResetTypeRegistry()` function to ensure that the type registry is in a deterministic state for tests, WDYT?